vela show --format markdown fails with ComponentDefinition

**Describe the bug**
with kubevela 1.9.5 the following error happens with ComponentDefinition:
```
vela show .webservice.cue --format markdown
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x48 pc=0x106b7aedc]

goroutine 1 [running]:
github.com/oam-dev/kubevela/references/docgen.(*MarkdownReference).GenerateMarkdownForCap(_, {_, _}, {{0x140010d8420, 0xa}, {0x106f1fa20, 0x13}, {0x140012be000, 0x36b0}, {0x0, ...}, ...}, ...)
	github.com/oam-dev/kubevela/references/docgen/markdown.go:174 +0x2fc
github.com/oam-dev/kubevela/references/docgen.(*MarkdownReference).CreateMarkdown(0x14000523ad0, {0x10800d0d8, 0x140000780c8}, {0x140008ce3c0, 0x1, 0x1}, {0x16b56f206, 0x4}, 0x0, 0x0?)
	github.com/oam-dev/kubevela/references/docgen/markdown.go:86 +0x1d4
github.com/oam-dev/kubevela/references/docgen.(*MarkdownReference).GenerateReferenceDocs(0x14000523ad0, {0x10800d0d8, 0x140000780c8}, {0x1400106c480, 0x0, 0x14000773490, {0x0, 0x0}, 0x14001018640, 0x0}, ...)
	github.com/oam-dev/kubevela/references/docgen/markdown.go:70 +0xe0
github.com/oam-dev/kubevela/references/cli.ShowReferenceMarkdown({0x10800d0d8, 0x140000780c8}, {0x0, 0x0, 0x14000773490, {0x0, 0x0}, 0x0, 0x0}, {{0x107fe6b00, ...}, ...}, ...)
	github.com/oam-dev/kubevela/references/cli/show.go:473 +0x124
github.com/oam-dev/kubevela/references/cli.NewCapabilityShowCommand.func1(0x14000331800?, {0x140010d51d0?, 0x5?, 0x5?})
	github.com/oam-dev/kubevela/references/cli/show.go:119 +0x334
github.com/spf13/cobra.(*Command).execute(0x14000331800, {0x140010d5180, 0x5, 0x5})
	github.com/spf13/cobra@v1.7.0/command.go:940 +0x5c8
github.com/spf13/cobra.(*Command).ExecuteC(0x1400029ef00)
	github.com/spf13/cobra@v1.7.0/command.go:1068 +0x35c
github.com/spf13/cobra.(*Command).Execute(0x107fe6b00?)
	github.com/spf13/cobra@v1.7.0/command.go:992 +0x1c
main.main()
	github.com/oam-dev/kubevela/references/cmd/cli/main.go:41 +0x174
```

This was not happening with kubevela 1.8 and it's working with traitDefinitions....

**To Reproduce**

Run vela show --format markdown with a ComponentDefinition as webservice

**Expected behavior**

Generate the markdown

**KubeVela Version**

1.9.5

**Cluster information**

Kubernetes 1.26

--- a/references/cli/show.go
+++ b/references/cli/show.go
@@ -464,12 +464,17 @@ func ShowReferenceConsole(ctx context.Context, c common.Args, ioStreams cmdutil.
 
 // ShowReferenceMarkdown will show capability in "markdown" format
 func ShowReferenceMarkdown(ctx context.Context, c common.Args, ioStreams cmdutil.IOStreams, capabilityNameOrPath, outputPath, location, i18nPath, ns string, rev int64) error {
-	ref := &docgen.MarkdownReference{}
-	paserRef, err := genRefParser(capabilityNameOrPath, ns, location, i18nPath, rev)
+	cli, err := c.GetClient()
 	if err != nil {
 		return err
 	}
-	ref.ParseReference = paserRef
+	ref := &docgen.MarkdownReference{}
+	parseRef, err := genRefParser(capabilityNameOrPath, ns, location, i18nPath, rev)
+	if err != nil {
+		return err
+	}
+	parseRef.Client = cli
+	ref.ParseReference = parseRef
 	if err := ref.GenerateReferenceDocs(ctx, c, outputPath); err != nil {
 		return errors.Wrap(err, "failed to generate reference docs")
 	}

